app/models/user.rb

```diff
@@ -525,6 +513,21 @@ def hotness_score
     search_score
   end
 
+  def connected_authentication_providers
+    identities_enabled.pluck(:provider).map(&:to_sym)
+  end
+
+  def authenticated_social_accounts
+    user_identities = identities_enabled
+    return {} if user_identities.blank?
+
+    urls = user_identities.pluck(:auth_data_dump)
+      .map { |data| data.dig(:info, :urls) }
+      .reduce(&:merge) || {}
+
+    { github: urls[:GitHub], twitter: urls[:Twitter], facebook: urls[:Facebook] }
```

Comment:
You may want to tag a `.compact` at the end so you can get rid of the `next if url.nil?` guard clause in the view.